packages/components/src/combobox-control/test/index.js

```diff
@@ -0,0 +1,141 @@
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ComboboxControl from '../';
+
+const timezones = [
+	{ name: 'Greenwich Mean Time', abbreviation: 'GMT' },
+	{ name: 'Universal Coordinated Time', abbreviation: 'UTC' },
+	{ name: 'European Central Time', abbreviation: 'ECT' },
+	{ name: '(Arabic) Egypt Standard Time', abbreviation: 'ART' },
+	{ name: 'Eastern African Time', abbreviation: 'EAT' },
+	{ name: 'Middle East Time', abbreviation: 'MET' },
+	{ name: 'Near East Time', abbreviation: 'NET' },
+	{ name: 'Pakistan Lahore Time', abbreviation: 'PLT' },
+	{ name: 'India Standard Time', abbreviation: 'IST' },
+	{ name: 'Bangladesh Standard Time', abbreviation: 'BST' },
+	{ name: 'Vietnam Standard Time', abbreviation: 'VST' },
+	{ name: 'China Taiwan Time', abbreviation: 'CTT' },
+	{ name: 'Japan Standard Time', abbreviation: 'JST' },
+	{ name: 'Australia Central Time', abbreviation: 'ACT' },
+	{ name: 'Australia Eastern Time', abbreviation: 'AET' },
+	{ name: 'Solomon Standard Time', abbreviation: 'SST' },
+	{ name: 'New Zealand Standard Time', abbreviation: 'NST' },
+	{ name: 'Midway Islands Time', abbreviation: 'MIT' },
+	{ name: 'Hawaii Standard Time', abbreviation: 'HST' },
+	{ name: 'Alaska Standard Time', abbreviation: 'AST' },
+	{ name: 'Pacific Standard Time', abbreviation: 'PST' },
+	{ name: 'Phoenix Standard Time', abbreviation: 'PNT' },
+	{ name: 'Mountain Standard Time', abbreviation: 'MST' },
+	{ name: 'Central Standard Time', abbreviation: 'CST' },
+	{ name: 'Eastern Standard Time', abbreviation: 'EST' },
+	{ name: 'Indiana Eastern Standard Time', abbreviation: 'IET' },
+	{ name: 'Puerto Rico and US Virgin Islands Time', abbreviation: 'PRT' },
+	{ name: 'Canada Newfoundland Time', abbreviation: 'CNT' },
+	{ name: 'Argentina Standard Time', abbreviation: 'AGT' },
+	{ name: 'Brazil Eastern Time', abbreviation: 'BET' },
+	{ name: 'Central African Time', abbreviation: 'CAT' },
+];
+const mapTimezoneOption = ( tz ) => ( {
+	value: tz.abbreviation,
+	label: tz.name,
+} );
+
+const timezoneOptions = timezones.map( mapTimezoneOption );
+const defaultLabelText = 'Select a timezone';
+const getLabel = () => screen.getByText( defaultLabelText );
+const getInput = () => screen.getByRole( 'combobox' );
+const getRenderedOptions = () => screen.getAllByRole( 'option' );
+
+describe( 'ComboboxControl', () => {
+	// Use real timers to avoid timeout errors from userEvent
+	beforeEach( () => {
+		jest.useRealTimers();
+	} );
+
+	const TestComboboxControl = ( props ) => {
+		const [ value, setValue ] = useState( null );
+		return (
+			<>
+				<ComboboxControl
+					{ ...props }
+					value={ value }
+					label={ defaultLabelText }
+					options={ timezoneOptions }
+					onChange={ setValue }
+				/>
+				<p data-testid="value-output">{ value }</p>
+			</>
+		);
+	};
+
+	it( 'should render', () => {
+		render( <TestComboboxControl /> );
+
+		const input = getInput();
+		expect( input ).toBeVisible();
```

Comment:
Interesting. My default assertion in this case is `toBeInTheDocument()`, but I also see why `toBeVisible` makes sense. Maybe we should add both, to make it more explicit when we later check for the label to be in the document, but not visible?

WDYT, @mirka ?

```suggestion
		expect( input ).toBeInTheDocument();
		expect( input ).toBeVisible();
```

Reply:
This makes sense to me, but I'll wait for @mirka's input as well 😄

Reply:
No strong opinion on the specifics, but I generally want the `it()` description to match what is actually being tested. "It should render" to me isn't a particularly useful description in this context. Should it say "It should render the input in the document" or "It should render a visible input in the document"? (But if that is actually what we do mean to test, I would find it more economical if that were tested as part of a longer interaction test, e.g. typing in the input and filtering options.)

Reply:
Thank you both... I've incorporated both bits of feedback now.

- I've gone ahead and removed the initial `should render` test, which was a pattern I was borrowing from another suite. I do agree that it's more economical to test that as part of other tests, which happens naturally anyway though!
- With that test removed, I did implement Marco's suggestion for presence of the label, and then its visibility.